moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -1060,13 +1063,41 @@ bool IKFastKinematicsPlugin::getPositionIK(const geometry_msgs::Pose& ik_pose, c
     ROS_DEBUG_STREAM_NAMED("ikfast", "No IK solution");
   }
 
+  // Among the solutions under limits, find the one that is closest to ik_seed_state
+  if (solutions_obey_limits.size() != 0)
+  {
+    double mindist = DBL_MAX;
+    int minindex = -1;
+    std::vector<double> sol;
+
+    for (size_t i = 0; i < solutions_obey_limits.size(); ++i)
+    {
+      sol = solutions_obey_limits[i];
+      double dist = harmonize(ik_seed_state, sol);
+      ROS_INFO_STREAM_NAMED("ikfast", "Dist " << i << " dist " << dist);
+
+      if (minindex == -1 || dist < mindist)
+      {
+        minindex = i;
+        mindist = dist;
+      }
+    }
+    if (minindex >= 0)
+    {
+      solution = solutions_obey_limits[minindex];
+      harmonize(ik_seed_state, solution);
+    }
+    error_code.val = moveit_msgs::MoveItErrorCodes::SUCCESS;
+    return true;
+  }
+
```

Comment:
Hi @jrgnicho ! What are the expectations from the getPositionIK call for multiple solutions? Am I being over-concerned about scanning the list of solutions in O(n) and not sorting them in n*logn?
I mean does this complexity make any difference? Because this was the only reason I didn't try sorting the elements. If it doesn't matter, I think your approach will be really beneficial and I will do that.